### PR TITLE
PPS: update of association cuts

### PR DESCRIPTION
**Pull request by jan-kaspar**
This PR adjusts the "association cuts" used in multi-RP proton reconstruction. This update has been presented in RECO/AT meeting:
https://indico.cern.ch/event/830302/contributions/3477928/attachments/1871174/3079130/jan_kaspar_pps.pdf
and PPS POG meeting:
https://indico.cern.ch/event/832270/contributions/3487106/attachments/1872773/3082425/jan_kaspar_pps.pdf

The new association cuts are different for 2016, 2017 and 2018 - which is achieved by introducing new era modifiers. For backward compatibility the meaning of  ``ctpps_2016`` is unchanged and this modifier still applies to all 2016, 17 and 18.

The association cuts are modified for all 2016, 2017 and 2018.

This update is backported to the 10_6 cycle for later stages of UL re-reco in #27475.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

File: RecoCTPPS/ProtonReconstruction/python/ctppsProtons_cff.py [hunk 1]
```diff
@@ -1,7 +1,47 @@
...
+  ctppsProtons.association_cuts_56.x_cut_apply = False
+  ctppsProtons.association_cuts_56.y_cut_apply = False
+  ctppsProtons.association_cuts_56.xi_cut_apply = True
+  ctppsProtons.association_cuts_56.xi_cut_value = 0.015
+  ctppsProtons.association_cuts_56.th_y_cut_apply = False
+
+ctpps_2016.toModify(ctppsProtons, ApplyDefaultSettings) # applied for all Run2 years (2016, 2017 and 2018)
+
+def Apply2017Settings(ctppsProtons):
+  ctppsProtons.association_cuts_45.xi_cut_value = 0.010
```
Comment (perrotta) [inline RecoCTPPS/ProtonReconstruction/python/ctppsProtons_cff.py]: Both these settings are identical to the "default" ones, i.e. you are not replacing anything here

File: RecoCTPPS/ProtonReconstruction/python/ctppsProtons_cff.py [hunk 2]
```diff
@@ -1,7 +1,47 @@
...
+
+# import and adjust proton-reconstructions settings
+from RecoCTPPS.ProtonReconstruction.ctppsProtons_cfi import *
 ctppsProtons.lhcInfoLabel = ctppsLHCInfoLabel
+
+from Configuration.Eras.Modifier_ctpps_2016_cff import ctpps_2016
+from Configuration.Eras.Modifier_ctpps_2017_cff import ctpps_2017
+from Configuration.Eras.Modifier_ctpps_2018_cff import ctpps_2018
+
+def ApplyDefaultSettings(ctppsProtons):
```
Comment (perrotta) [inline RecoCTPPS/ProtonReconstruction/python/ctppsProtons_cff.py]: Start with a lowercase letter, as for cms code rules: ```applyDefaultSettings``` (and the same for the similar cases below)

**Comment by perrotta** [comment]
By the way, the only difference visible in the jenkins tests is in the 2016 workflow 136.731, where an additional proton is reconstructed with the new settings.
This demonstrates that you are also modifying the setting for 2016 (where before this PR you were just relying on those set by the fillDescriptions method of CTPPSProtonProducer.cc, which are different from the ones set here with the ctpps_2016 modifier). I think that this is exactly what you want, but let me underline it here because it was not obvious from the PR description.

I imagine that the automated comparisons do not show differences in the 2017 or 2018 workflows just because of the low statistics.

File: Configuration/Eras/python/Era_Run2_2018_cff.py [hunk 3]
```diff
@@ -15,7 +15,8 @@
 from Configuration.Eras.Modifier_run2_egamma_2018_cff import run2_egamma_2018
 from Configuration.Eras.Modifier_run2_egamma_2017_cff import run2_egamma_2017
 from Configuration.Eras.Modifier_run2_L1prefiring_cff import run2_L1prefiring
+from Configuration.Eras.Modifier_ctpps_2018_cff import ctpps_2018
 
 Run2_2018 = cms.ModifierChain(Run2_2017.copyAndExclude([run2_HEPlan1_2017, run2_muon_2017, run2_L1prefiring, run2_HLTconditions_2017, run2_egamma_2017]),
```
[the remark this answers is not in the document]
Comment (jan-kaspar) [reply Configuration/Eras/python/Era_Run2_2018_cff.py]: Absolutely, thanks for catching this! Fixed in d8eb24c.

File: RecoCTPPS/ProtonReconstruction/python/ctppsProtons_cff.py (hunk 1 above)
Comment (jan-kaspar) [reply RecoCTPPS/ProtonReconstruction/python/ctppsProtons_cff.py]: Right, in the POG meeting we converged on identical settings for 2016 and 2017. For symmetry (and possible future updates) I kept the code separated for 2016 and 2017.

File: RecoCTPPS/ProtonReconstruction/python/ctppsProtons_cff.py (hunk 2 above)
Comment (jan-kaspar) [reply RecoCTPPS/ProtonReconstruction/python/ctppsProtons_cff.py]: Fixed in d8eb24c.

**Comment by jan-kaspar** [comment]
@perrotta Thanks for your check. I attach a plot of my private validation where changes in 2016, 17 and 18 are visible.

make_cmp png-000001

I've modified the PR description to explicitly mention that the settings are modified for all 2016, 17 and 18.

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- Parameters for association cuts are updated separately for the last three years of data taking
- Small changes with respect to the previous common default are shown in https://github.com/cms-sw/cmssw/pull/27440#issuecomment-509727882 :  jenkins tests pass and show no other differences besides the low-stat reflection of these changes.

**Comment by perrotta** [comment]
Please @jan-kaspar prepare the backpport to 10_6 that you have planned

**Comment by jan-kaspar** [comment]
Thanks @perrotta, the backport is here: #27475.

**Comment by fabiocos** [comment]
+operations

the update of Configuration eras is coherent with the purpose of the PR

**Comment by fabiocos** [comment]
+1

**Pull request merged**